src/EditorFeatures/CSharp/Formatting/TypingFormattingRule.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Formatting;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp.Utilities;
+using Microsoft.CodeAnalysis.Formatting.Rules;
+using Microsoft.CodeAnalysis.Options;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.Formatting
+{
+    internal class TypingFormattingRule : BaseFormattingRule
```

Comment:
i don't really understand why this is called TypingFormattingRule .  There def needs to be a common this explaining what it is needed for.

Reply:
well, it is random name, if you want better name, you can suggest one.

Reply:
I don't know what to suggest because i don't know what the purpose of the change was.  Hence why i asked:

> this code needs clear explanations about waht it is tring to do and why it is appropriate. Why do we include the TypingFormattingRule.Instance if we're not on CloseBrace/EOF?